### DISCUSSION TITLE
Console output stops pretty printing, reverts to black and white

### DISCUSSION THREAD

**Discussion by turbanisch** [ropensci/targets#1209] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Whenever I use the targets package, at some point the R Studio console will stop pretty printing (e.g., using different colors for printing tibbles, see images below). Unfortunately, I can't pinpoint exactly what triggers this behavior – in the example below, it seems to occur after running `tar_source()`. I usually have the Tidyverse packages loaded, as well as targets and tarchetypes (`tidyverse_2.0.0 tarchetypes_0.7.11 targets_1.4.1 `). 

Here is some of my session info:

R version 4.3.1 (2023-06-16)
R Studio 2023.12.0 Build 369
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.2.1

<img width="358" alt="grafik" src="https://github.com/ropensci/targets/assets/51969412/112a49ba-1728-4510-ae5d-a2834b0a9477">

It's a minor issue for sure but something that bugs me personally because I rely very much on visual clues such as NA values printed in red, etc. Any hints are appreciated! 🥳

**Comment by mattwarkentin** [comment]
FWIW, I have the same thing happen to my R console (in RStudio) after I `tar_make()` my pipeline. I am also not sure what the issue is.

**Reply by benzipperer** [reply]
I'm so happy to see I'm not the only one with this problem! For me, what triggers the disappearing colors problem is running a pipeline that produces a warning or an error message. It occurs in the R console in Rstudio or in a terminal.

For example, consider the following pipeline that errors out
``` r
targets::tar_dir({
  targets::tar_script({
    library(targets)
    list(tar_target(model, lm(mpg ~ weight, data = mtcars)))
  })
  targets::tar_make()
})
#> ▶ dispatched target model
#> ✖ errored target model
#> ✖ errored pipeline [0.065 seconds]
#> Error:
#> ! Error running targets::tar_make()
#> Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
#> Debugging guide: https://books.ropensci.org/targets/debugging.html
#> How to ask for help: https://books.ropensci.org/targets/help.html
#> Last error message:
#>     object 'weight' not found
#> Last error traceback:
#>     lm(mpg ~ weight, data = mtcars)
#>     eval(mf, parent.frame())
#>     eval(mf, parent.frame())
#>     stats::model.frame(formula = mpg ~ weight, data = mtcars, drop.unused.le...
#>     model.frame.default(formula = mpg ~ weight, data = mtcars, drop.unused.l...
#>     eval(predvars, data, env)
#>     eval(predvars, data, env)
#>     .handleSimpleError(function (condition)  {     state$error <- build_mess...
#>     h(simpleError(msg, call))
```

<sup>Created on 2024-01-23 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

After running that pipeline, I lose most colors in the console:
![Screenshot from 2024-01-23 10-59-07](https://github.com/ropensci/targets/assets/4070964/2affe907-b11d-458c-96dc-10438f7579ae)

But before running that pipeline, the colors are fine:
![Screenshot from 2024-01-23 11-00-31](https://github.com/ropensci/targets/assets/4070964/db3cb414-4c6f-4e26-ba00-e357e2a5e4a6)

Can others reproduce this?

**Comment by wlandau** [comment]
Should be fixed as of 670b028daa0346b38eb2e175d2d08eef4740e923, c.f. https://github.com/ropensci/targets/issues/1210#issuecomment-1906397463.

**Reply by benzipperer** [reply]
ahhh! oops, I missed that issue in my search. THANK YOU! 

this solves my problem
